### PR TITLE
Bookmark songs

### PR DESCRIPTION
**Pull request by ninapeters**
Hi there! Please check out my app. I appreciate every code review and comment.
I have adjusted the file structure. It's now possible to save a song and switch between the two lists.

### CODE REVIEW

File: src/components/Buttons/Button.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  handleClick: PropTypes.func,
+  children: PropTypes.string.isRequired,
+}
+
+export default function Button({ disabled, handleClick, children }) {
+  return (
+    <ButtonStyled
+      disabled={disabled}
+      onClick={handleClick}
+      data-testid="button"
```
Comment (omorgenroth) [inline src/components/Buttons/Button.js:15]: Do you use that data-testid anywhere in your tests right now? Couldn`t find it. Otherwise maybe just leave it out for now.

File: src/components/Buttons/AudioButton.js [hunk 2]
```diff
@@ -5,12 +5,12 @@ import { ReactComponent as PauseIcon } from '../../assets/pause.svg'
...
 AudioButton.propTypes = {
   handleClick: PropTypes.func,
-  isSongPlaying: PropTypes.bool,
+  isSongPlaying: PropTypes.bool.isRequired,
 }
 
 export default function AudioButton({ handleClick, isSongPlaying }) {
   return (
-    <Button onClick={handleClick}>
+    <Button onClick={handleClick} data-testid="audio-button">
```
Comment (omorgenroth) [inline src/components/Buttons/AudioButton.js:13]: Same here with the data-testid.

File: src/components/Navigation/Navigation.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  return <NavButton onClick={handleClick}>{children}</NavButton>
+}
+
+const NavButton = styled.button`
+  background-color: transparent;
+  border: var(--primary-dark) 2px solid;
+  color: var(--primary-dark);
+  padding: 6px;
+  font-size: 0.8em;
+`
```
Comment (omorgenroth) [inline src/components/Navigation/Navigation.js:19]: Could you not just reuse your Button Component here? Styling looks similar

File: src/components/App.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+`
+const Footer = styled.footer`
+  place-self: center;
+`
+const UnmuteMessage = styled.p`
+  padding: 4px;
+  color: var(--primary-dark-transparent);
+  text-align: center;
+  font-size: 0.7em;
+`
```
Comment (omorgenroth) [inline src/components/App.js:81]: Why did you move your App.js to your Components Folder? That´s a bit confusing to me.

File: src/components/App.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
+import styled from 'styled-components'
```
Comment (JanaHaeusler) [inline src/components/App.js:1]: /macro

File: src/components/Buttons/Button.test.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+      </Button>
+    )
+    expect(container.firstChild).toMatchSnapshot()
+
+    rerender(
+      <Button disabled={true} handleClick={() => {}}>
+        Save this song
+      </Button>
+    )
+    expect(container.firstChild).toMatchSnapshot()
```
Comment (JanaHaeusler) [inline src/components/Buttons/Button.test.js:19]: Maybe separate render test into two tests - one for disabled and one for enabled. Test could fail for two reasons.

File: src/components/Navigation/Navigation.test.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
+import { render } from '@testing-library/react'
+import Navigation from './Navigation'
+import user from '@testing-library/user-event'
+
+describe('Navigation', () => {
+  it('renders correctly', () => {
+    render(<Navigation onClick={() => {}}>all songs</Navigation>)
+  })
```
Comment (JanaHaeusler) [inline src/components/Navigation/Navigation.test.js:8]: Did you intend to run a snapshot test? It is not testing at the moment, isn't? Just rendering?

File: src/components/Pages/ArtistPage/ArtistPage.js [hunk 8]
```diff
@@ -1,29 +1,46 @@
...
-import AudioButton from '../Audio/AudioButton'
-import useAudio from '../Audio/useAudio'
+import AudioButton from '../../Buttons/AudioButton'
+import Button from '../../Buttons/Button'
 
 ArtistList.propTypes = {
   artists: PropTypes.array.isRequired,
+  toggleCurrentSongId: PropTypes.func.isRequired,
+  isSongPlaying: PropTypes.bool.isRequired,
+  currentSongId: PropTypes.string,
```
Comment (JanaHaeusler) [inline src/components/Pages/ArtistPage/ArtistPage.js:10]: Is it required?

File: src/components/Pages/ArtistPage/ArtistPage.js [hunk 9]
```diff
@@ -1,29 +1,46 @@
...
+  saveSong: PropTypes.func.isRequired,
+  savedSongs: PropTypes.array,
```
Comment (JanaHaeusler) [inline src/components/Pages/ArtistPage/ArtistPage.js:12]: Is it required?

File: src/components/Pages/ArtistPage/ArtistPage.test.js [hunk 10]
```diff
@@ -0,0 +1,165 @@
...
+        currentSongId="a"
+        saveSong={saveSongMock}
+        savedSongs={[]}
+      />
+    )
+    const button = getByTestId('button')
+    user.click(button)
+    expect(saveSongMock).toHaveBeenCalled()
+  })
+  it('renders a new array for savedSongs by calling saveSong function', () => {
```
Comment (JanaHaeusler) [inline src/components/Pages/ArtistPage/ArtistPage.test.js:117]: I am little bit confused by this test, maybe I just do not get it though. :D 
You want to test, if a new array is rendered but you fill it yourself?

File: src/components/hooks/useAudio.js [hunk 11]
```diff
@@ -1,12 +1,13 @@
...
 
-export default function useAudio({ artists }) {
+export default function useAudio({ artistData }) {
   const [currentSongId, setCurrentSongId] = useState()
   const [isSongPlaying, setIsSongPlaying] = useState(false)
   const songRef = useRef(null)
 
-  const songUrl = artists.find((artist) => artist.id === currentSongId)?.songUrl
+  const songUrl = artistData.find((artist) => artist.id === currentSongId)
+    ?.songUrl
```
Comment (JanaHaeusler) [inline src/components/hooks/useAudio.js:10]: A bit confusing. :D Why is it in an extra line?

**Review by JanaHaeusler — COMMENTED** [review]
Love all your hard coding work! Amazig app, which you are programming and detailed tests. I left some comments and thoughst. Hopefully they are of any help. Feel free to ask, in case something is not comprehensible. Happy coding! :)

File: src/components/App.js (hunk 4 above)
Comment (ninapeters) [reply src/components/App.js:81]: To me it is clearer to only have the index.js as a global file.

File: src/components/Buttons/AudioButton.js (hunk 2 above)
Comment (ninapeters) [reply src/components/Buttons/AudioButton.js:13]: I need the data-testid to test individual button functions in my ArtistList.

File: src/components/Buttons/Button.js (hunk 1 above)
Comment (ninapeters) [reply src/components/Buttons/Button.js:15]: same here :)

File: src/components/Buttons/Button.test.js (hunk 6 above)
Comment (ninapeters) [reply src/components/Buttons/Button.test.js:19]: I think in this case there is no need to split the test as it is a simple snapshot test.

File: src/components/Navigation/Navigation.js (hunk 3 above)
Comment (ninapeters) [reply src/components/Navigation/Navigation.js:19]: To structure my code clearly, I think it's better to separate a regular call to action button from a navigation button.

File: src/components/Navigation/Navigation.test.js (hunk 7 above)
Comment (ninapeters) [reply src/components/Navigation/Navigation.test.js:8]: Thanks for the hint! :D

File: src/components/hooks/useAudio.js (hunk 11 above)
Comment (ninapeters) [reply src/components/hooks/useAudio.js:10]: Prettier. I will try to fix it.
